src/libraries/wizard/relationslabelscontroller.cpp

```diff
@@ -102,16 +102,21 @@ void appendModuleItems(const QList<Metadata::Reference> &references, QTreeWidget
 }
 } // namespace Anonymous
 
-void RelationsLabelsController::onFocusedItemChanged(const QModelIndex &current, const QModelIndex &previous) const
+void RelationsLabelsController::onSelectionChanged(const QItemSelection &selected, const QItemSelection &deselected)
 {
-    Q_UNUSED(previous);
+    Q_UNUSED(deselected);
 
-    if (!current.isValid())
-        return;
+    updateRelationsTrees(selected.indexes().value(0));
+}
 
+void RelationsLabelsController::updateRelationsTrees(const QModelIndex &current)
+{
     m_requiresTree->clear();
     m_conflictsTree->clear();
 
+    if (!current.isValid())
+        return;
+
```

Comment:
imho it would be a little clearer to do "clear" in "onSelectionChanged" (maybe even  extract it to separate function). Now this "!current.Valid" looks surprising in method named "update RelationsTrees. What I mean is - "onSelectionChanged" == clearTrees && rebuildTrees. And in "rebuildTrees" -> if !valid -> exit early.

Reply:
Agreed.